### DISCUSSION TITLE
Add an option to display custom OpenAPI Extensions properties

### DISCUSSION THREAD

**Discussion by obrassard** [scalar/scalar#3825] [Ideas]
Hi ! I think it would be nice if Scalar Api Reference could display custom [Open API extensions](https://swagger.io/docs/specification/v3_0/openapi-extensions/) properties as Swagger UI does.

In my team, we use these properties in particular to document which user roles are required to access a specific endpoint. But I've seen different use cases for these properties on the web.

In swagger UI this is how the information is presented, but i'm sure there would be a way to display those properties in a nicer way 

<img width="775" alt="Capture d’écran, le 2024-11-08 à 13 58 54" src="https://github.com/user-attachments/assets/9ff29168-7e3c-4f22-9808-f0f9cc2946e1">

It would be nice to be able to specify which extensions properites you want to show as additional documentation on endpoints, and for this information to be displayed along with other endpoint data such as headers, parameters, responses, etc.

**Comment by ludovic-pourrat** [comment]
It would be an awesome feature !

**Comment by SheedGuy** [comment]
Chiming in here to say this would be very helpful.

**Comment by hanspagel** [comment]
we do have the plugin API 
https://scalar.com/products/api-references/plugins
